📝 Add Word Types Chapter

Benjamin-Piper opened the pull request:
To build locally, please change chapter on line 2 of `03_word_types.mdx` to 2.

Blocked by #1 for styling

Please see the commit descriptions

Benjamin-Piper committed:
➕ Add React

Benjamin-Piper committed:
➕ Add React Flow

One of the main reasons for using Astro was components. I want
easily-maintainable diagrams.

Maintaining text via Git is much easier than images.

Benjamin-Piper committed:
✨ Add WordTypes component

Benjamin-Piper committed:
📝 Add Word Types Chapter

Both The Alphabet and this chapter work to cover the whole
morphology.

The pairs image was included  as is and not rewritten as a table
because it's unlikely to change.

mia-entropy commented on src/content/chapters/03_word_types.mdx:
```diff
@@ -0,0 +1,181 @@
+- TODO INSERT TRANSITIVE: 
+- gli: relation for "_ is happy that _ is true".
+
+### Compounds
```
You should explain borrowings first, then tell what can appear inside compounds (root, particles, and a borrowing in first position only)

mia-entropy commented on src/content/chapters/03_word_types.mdx:
```diff
@@ -0,0 +1,181 @@
+### Borrowings
+
+Borrowings let you import foreign words or names. They are not present in the
+dictionary, but can appear in compounds combined with a root word.
```
```suggestion
dictionary, but can appear in compounds combined with a root word or particle.
```

mia-entropy commented on src/content/chapters/03_word_types.mdx:
```diff
@@ -0,0 +1,181 @@
+   1. A single letter (yes this includes h)
+   1. An initial or medial pair
+   1. A consonant triplet
+1. Optionally continue alternating between vhowels and consonant pairs/triplets.
```
Sonorants and single consonants are also allowed in this position.

mia-entropy commented on src/content/chapters/03_word_types.mdx:
```diff
@@ -0,0 +1,181 @@
+- euinglicban: property for "_ is the English language".
+- za u'unicef: property for "_ is named UNICEF".
+
+Borrowings are terminated with the particle `be`, but this is only required when
```
Maybe give more details about that. 2 consecutive borrowings not separated with `be` merge together:
- za udjon ufordo = E is named John Ford
- za udjon be ufordo = E is named John and is "a ford" (maybe the car brand?).

mia-entropy commented on src/content/chapters/03_word_types.mdx:
```diff
@@ -0,0 +1,181 @@
+Examples:
+
+- eufransestan: property for "_ is France".
+- euinglicban: property for "_ is the English language".
```
Borrowing must be ended with ' or space inside compounds too, as ufransestan is a valid borrowing. 
```suggestion
- eufranse'stan: property for "_ is France".
- euinglic'ban: property for "_ is the English language".
```

mia-entropy commented on src/content/chapters/03_word_types.mdx:
```diff
@@ -0,0 +1,181 @@
+variables do not appear in the dictionary at all.
+
+Freeform variables follow the same structure as borrowings except that their
+prefix is "i". Here's an example:
```
Add that they don't merge together and thus cannot be terminated with `be`, which would collide with usage of `be` at the end of an argument list.

Benjamin-Piper replied:
I'll mention `be` used as an argument list terminator in the bindings section.

Benjamin-Piper committed:
💄 Add Examples type to Info.astro

Benjamin-Piper committed:
further revisions

Benjamin-Piper committed:
💄 Adjust dark mode styling of Info.astro

White text wasn't super legible. I tried hard to get white text on yellow
to work but settled for black in the end. It makes the warning all the
more jarring anyway.

Benjamin-Piper commented:
Henceforth I'll be merging without approvals. It's easier to read the finished website and then make a PR anyway.
